File: solutions/go/rna-transcription/1/rna_transcription.go
// Package strand should have a package comment that summarizes what it's about.
package strand

// ToRNA should have a comment documenting it.
func ToRNA(dna string) string {
	complement := map[rune]string{
		'G': "C",
		'C': "G",
		'T': "A",
		'A': "U",
	}
	rna := ""
	for _, nucleotide := range dna {
		rna += complement[nucleotide]
	}
	return rna
}

File: solutions/cpp/rna-transcription/1/rna_transcription.cpp
#include "rna_transcription.h"

namespace rna_transcription {

char to_rna(const char dna)
{
    switch (dna) {
    case 'C':
        return 'G';
    case 'G':
        return 'C';
    case 'A':
        return 'U';
    case 'T':
        return 'A';
    default:
        return '\0';
    }
}

std::string to_rna(const std::string dna)
{
    std::string rna{};
    for (const auto& nucleotide : dna) {
        rna.push_back(to_rna(nucleotide));
    }
    return rna;
}

}  // namespace rna_transcription
